horaedb/metric_engine/src/types.rs

```diff
@@ -0,0 +1,51 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::{ops::Range, pin::Pin, sync::Arc};
+
+use arrow::{array::RecordBatch, datatypes::Schema};
+use futures::Stream;
+use object_store::ObjectStore;
+
+use crate::error::Result;
+
+pub enum Value {
+    Int64(i64),
+    Int32(i32),
+    Float64(f64),
+    Bytes(Vec<u8>),
+}
+
+pub enum Predicate {
```

Comment:
The `TimeMergeStorage` is low level trait for operating arrow and parquet, I think maybe we just use filter expr in `arrow` for flexibility?

Reply:
Expr belong to DataFusion, not arrow.